morpheus/pipeline/pipeline.py

```diff
@@ -96,7 +96,7 @@ def _add_id_col(self, x: cudf.DataFrame):
 
         return x
 
-    def add_node(self, node: StreamWrapper, segment_id: str = "main"):
+    def _add_node(self, node: StreamWrapper, segment_id: str = "main"):
```

Comment:
`add_node` is a public method. It could possibly be renamed `add_stage`. It will be used by non-linear pipelines